pkg/storage/tsdb/block/block_test.go

```diff
@@ -414,6 +414,59 @@ func TestMarkForNoCompact(t *testing.T) {
 	}
 }
 
+func TestUnMarkForNoCompact(t *testing.T) {
+	testutil.VerifyNoLeak(t)
+	ctx := context.Background()
+	tmpDir := t.TempDir()
+	for _, tcase := range []struct {
+		name          string
```

Comment:
unless you need strict ordering, I would find `map[string]struct{..}` nicer, with key being the test name.

Reply:
I was using the vector to keep the similar test styles with all other tests in this file. I don't have strong opinion on this, will change it to map.